src/asymmetric.rs

```diff
@@ -26,14 +26,13 @@ unsafe extern "C" fn p_sign(
     _signature_size: usize,
     p_signature_length: *mut usize,
 ) -> psa_status_t {
-    let hash = std::slice::from_raw_parts(p_hash, hash_length).to_vec();
     let alg = match AsymmetricSignature::try_from(alg) {
         Ok(alg) => alg,
         Err(e) => return e.into(),
     };
     let signature = match PARSEC_BASIC_CLIENT.read().unwrap().psa_sign_hash(
         key_slot_to_key_name(key_slot),
-        hash,
+        std::slice::from_raw_parts(p_hash, hash_length),
```

Comment:
I'm guessing we're bound to fully trust the caller on this one? From the documentation, `from_raw_parts` seems to have quite a few safety pitfalls. But since these methods are already tagged `unsafe`, I see no reason to object.

Reply:
I guess we assume that the PSA Crypto implementation that uses this driver is trusted 😬